src/main/java/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package bookstore.service.impl;
+
+import bookstore.dto.order.OrderCreateDto;
+import bookstore.dto.order.OrderItemResponseDto;
+import bookstore.dto.order.OrderResponseDto;
+import bookstore.dto.order.OrderUpdateDto;
+import bookstore.entity.order.Order;
+import bookstore.entity.user.User;
+import bookstore.mapper.OrderItemMapper;
+import bookstore.mapper.OrderMapper;
+import bookstore.repository.OrderItemRepository;
+import bookstore.repository.OrderRepository;
+import bookstore.repository.UserRepository;
+import bookstore.service.OrderService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final UserRepository userRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrder(OrderCreateDto createDto, String username) {
+        return orderMapper.toDto(
+                orderRepository.save(
+                        orderMapper.toModel(
+                            userRepository.getUserByEmail(username), createDto
+                        )
+                )
+        );
+    }
```

Comment:
Correct me, if I'm wrong, but this way you create empty order for authenticated user. Thus, there are no orderitems in it. Then how do you actually add order items from the shopping cart here? I can't seem to find such implementation in your code. 

Don't mind this comment, if I'm missing something and everything works as intended